src/rgw/rgw_lua_request.cc

```diff
@@ -342,14 +342,80 @@ struct BucketMetaTable : public EmptyMetaTable {
   }
 };
 
+rgw::sal::Object* get_object_with_atttributes(const req_state* s, rgw::sal::Object* obj) {
+  // in case of copy obj, the tags and metadata are taken from source
+  const auto src_obj = s->src_object ? s->src_object.get() : obj;
```

Comment:
> > they could only be used in postrequest context
> 
> if these attributes are only valid in postrequest, why read attributes from the source object? wouldn't the same attrs be present in `s->object` during postrequest?
> 
ok, i can change that to the object

> i don't think it's a good idea to re-read anything from the backend here. there's no guarantee that we'd read from the same upload that the op copied from, because the source object may have been overwritten since

my problem is that in the case of "put_obj" the existing attributes are not bein read from the backend

Reply:
will make sure that the lack of atomicity is documented.
any idea of how should i call the existing object, so there is no ambiguity?
should I go with `Request.StoredObject...` ?